src/jit/gentree.cpp

```diff
@@ -14705,11 +14705,26 @@ void Compiler::gtCheckQuirkAddrExposedLclVar(GenTreePtr tree, GenTreeStack* pare
 #endif
 }
 
-// Checks to see if we're allowed to optimize Type::op_Equality or Type::op_Inequality on this operand.
-// We're allowed to convert to GT_EQ/GT_NE if one of the operands is:
-//  1) The result of Object::GetType
-//  2) The result of typeof(...)
-//  3) a local variable of type RuntimeType.
+//------------------------------------------------------------------------
+// gtCanOptimizeTypeEquality: check if tree is a suitable input for a type
+//    equality optimization
+//
+// Arguments:
+//    tree - tree to examine
+//
+// Return Value:
+//    True, if value of tree can be used to optimize type equality tests
+//
+// Notes:
+//    Checks to see if the jit is able to optimize Type::op_Equality or
+//    Type::op_Inequality calls that consume this tree.
+//
+//    The jit can safely convert these methods to GT_EQ/GT_NE if one of
+//    the operands is:
+//    1) The result of Object::GetType
+//    2) The result of typeof(...)
+//    3) Is otherwise known to have type RuntimeType
```

Comment:
The "otherwise" here got me wondering -- should `gtGetClassHandle` itself identify the first two cases as producing type `RuntimeType`, either from their signature or from moving this special casing into it?

Reply:
Ultimately, yes. 

However, I've been holding off adding new cases to `gtGetClassHandle` as I am expecting to have a more general solution for tracking ref class tree node types.

Reply:
SGTM